app/services/foreman/renderer/scope/macros/base.rb

```diff
@@ -112,12 +112,8 @@ def pxe_kernel_options
             example "save_to_file('/etc/motd', \"hello\\nworld\\n\") # => 'cat << EOF > /etc/motd\\nhello\\nworld\\nEOF'"
           end
           def save_to_file(filename, content)
-            content = content.lines.map { |line| ' ' + line }.join()
-            if !content || content.ends_with?("\n")
-              "cat << EOF | sed 's/^ //' > #{filename}\n#{content}EOF"
-            else
-              "cat << EOF | sed 's/^ //' > #{filename}\n#{content}\nEOF"
-            end
+            encoded = Base64.encode64(content || '')
+            "cat << EOF | base64 -d > #{filename}\n#{encoded}EOF"
```

Comment:
This relies on `Base64.encode64` returning with a line ending, but on an empty string it doesn't:
```
[1] pry(main)> require 'base64'
=> true
[2] pry(main)> Base64.encode64('bla')
=> "Ymxh\n"
[3] pry(main)> Base64.encode64('')
=> ""
```

Perhaps you could generate `echo > #{filename}` if `encoded.empty?` to make it very obvious the file is being emptied.